lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -384,6 +398,34 @@ data ApiAddress (n :: NetworkDiscriminant) = ApiAddress
     , state :: !(ApiT AddressState)
     } deriving (Eq, Generic, Show)
 
+newtype ApiScript = ApiScript
+    { script :: ApiT Script
+    } deriving (Eq, Generic, Show)
+
+newtype ApiPubKey = ApiPubKey
+    { pubKey :: ByteString
+    } deriving (Eq, Generic, Show)
+
+data Credential = Credential
+    { credential :: Either ApiPubKey ApiScript
+    } deriving (Eq, Generic, Show)
+
+data ApiCredentials = ApiCredentials
+    { spending :: !(Maybe Credential)
+    , staking :: !(Maybe Credential)
+    } deriving (Eq, Generic, Show)
+
+data AnyAddressType =
+      EnterpriseDelegating
+    | RewardAccount
+    deriving (Eq, Show, Bounded, Enum)
+
+data AnyAddress = AnyAddress
+    { payload :: ByteString
+    , flavour :: AnyAddressType
+    , network :: Int
+    } deriving (Eq, Generic, Show)
```

Comment:
Why not re-use the existing types for Addresses ?

Reply:
If I use Addresses I will be able to differentiate between `addr` and `addr_test` based on network. but I will not be able to use reward_account's `stake` and `stake_test` (to my best knowledge). This was main reason I introduced this type